examples/regression/Ridge2FoldCVRegularization.py

```diff
@@ -0,0 +1,374 @@
+# %%
+
+r"""
+ RidgeRegression2FoldCV for data with low effective rank
+ =======================================================
+ In this notebook we explain in more detail how
+ :class:`skmatter.linear_model.RidgeRegression2FoldCV` speeds up the
+ cross-validation optimizing the regularitzation parameter :param alpha: and
+ compare it with existing solution for that in scikit-learn
+ :class:`slearn.linear_model.RidgeCV`.
+ :class:`skmatter.linear_model.RidgeRegression2FoldCV` was designed to predict
+ efficiently feature matrices, but it can be also useful for the prediction
+ single targets.
+"""
+# %%
+#
+
+import time
+
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.datasets import make_regression
+from sklearn.linear_model import RidgeCV
+from sklearn.metrics import mean_squared_error
+from sklearn.model_selection import KFold, train_test_split
+
+from skmatter.linear_model import RidgeRegression2FoldCV
+
+
+# %%
+
+SEED = 12616
+N_REPEAT_MICRO_BENCH = 5
+
+# %%
+# Comparing skmatter 2-fold CV with sklearn leave-one-out CV
+# ----------------------------------------------------------
+#
+# In linear regression, the complexity of computing the weight matrix is
+# theoretically bounded by the inversion of the covariance matrix.  This is
+# more costly when conducting regularized regression, wherein we need to
+# optimise the regularization parameter in a cross-validation (CV) scheme,
+# thereby recomputing the inverse for each parameter.  scikit-learn offers an
+# efficient leave-one-out CV (LOO CV) for its ridge regression which avoids
+# these repeated computations [loocv]_. Because we needed an efficient ridge that works
```

Comment:
might be nicer to directly link the slides here somewhere (as well as leaving a reference at the bottom)